libs/BLEKit/include/BLEKit.h

```diff
@@ -22,12 +22,19 @@ class BLEKit
 
 	void init();
 
+	void setName(const char *name);
+	void setBatteryLevel(uint8_t value);
+	void setIsCharging(bool value);
+	void setDisplayedColor(uint8_t red_channel, uint8_t green_channel, uint8_t blue_channel);
+
```

Comment:
Having BLEKit know about name, battery, charging state and displayed color breaks the SRP.

To fix that, you have two methods:

- `void setAdvertisingData(AdvertisingData & data)`
- `auto getAdvertisingData() -> AdvertisingData`

This way, the `AdvertisingData` struct can evolve without impacting BLEKit

Reply:
Yes, but not sure which gonna store it, BLEKit or Coregap. In the last case, BLEKit will be only a passthrough.

Reply:
it cas be in CoreGap then